mavros/include/mavros/plugin.hpp

```diff
@@ -110,7 +101,7 @@ class Plugin : public std::enable_shared_from_this<Plugin>
   }
 
 protected:
-  UASPtr uas;                       // uas back link
+  std::weak_ptr<UAS> uas_;                       // uas back link
```

Comment:
What for? Shared ptr good enough.

Reply:
`shared_ptr` is not good here, because it creates cyclic references. The UAS node holds `shared_ptr`s to plugin nodes. In turn, each plugin node holds a `shared_ptr` to the UAS node. The result is that the UAS node and the plugin nodes never get destroyed.
You can check this by loading the `mavros::uas::UAS` component into a component container and then unloading it. The nodes will still be there. And then load it again, every node will get duplicated.